### PR TITLE
Support listConcatenateByRows in Java package [skip ci]

### PR DESCRIPTION
**Pull request by sperlingxx**
Current PR is to provide Java API for `cudf::lists::concatenate_rows`, which is added in #8049.

### CODE REVIEW

File: java/src/main/native/src/ColumnViewJni.cpp [hunk 1]
```diff
@@ -1049,6 +1050,28 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_stringConcatenation(
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_listConcatenationByRow(JNIEnv *env,
+                                                                              jobject j_object,
```
[the remark this answers is not in the document]
Comment (sperlingxx) [reply java/src/main/native/src/ColumnViewJni.cpp]: Fixed.

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 2]
```diff
@@ -2849,6 +2849,16 @@ private static native long stringReplaceWithBackrefs(long columnView, String pat
...
 
+
+  /**
+   * Native method to concatenate columns of lists together, combining a row from
+   * each column into a single list.
+   * @param columnViews array of longs holding the native handles of the column_views to combine.
+   * @return native handle of the resulting cudf column, used to construct the Java column
+   *         by the listConcatenateByRow method.
+   */
+  protected static native long listConcatenationByRow(long[] columnViews, boolean ignoreNull);
```
[the remark this answers is not in the document]
Comment (sperlingxx) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java]: I just followed the style of `stringConcatenation` without much thought. I changed it as a private static method of `ColumnVector`.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 3]
```diff
@@ -527,6 +527,48 @@ public static ColumnVector stringConcatenate(Scalar separator, Scalar narep, Col
...
+    long size = columns[0].getRowCount();
+    long[] columnViews = new long[columns.length];
+    DType childType = columns[0].getChildColumnView(0).getType();
+    assert !childType.isNestedType() : "listConcatenateByRow only supports lists with depth 1";
+
+    for(int i = 0; i < columns.length; i++) {
+      assert columns[i] != null : "Column vectors passed may not be null";
+      assert columns[i].getType().equals(DType.LIST) : "All columns must be of type list";
+      assert columns[i].getRowCount() == size : "Row count mismatch";
+      assert childType.equals(columns[i].getChildColumnView(0).getType()) : "Element type mismatch";
```
[the remark this answers is not in the document]
Comment (sperlingxx) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Fixed, thank you.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 4]
```diff
@@ -527,6 +527,48 @@ public static ColumnVector stringConcatenate(Scalar separator, Scalar narep, Col
     return new ColumnVector(stringConcatenation(column_views, separator.getScalarHandle(), narep.getScalarHandle()));
   }
 
+  /**
+   * Concatenate columns of lists horizontally (row by row), combining a corresponding row
+   * from each column into a single list row of a new column. Null list element of input columns
+   * will be ignored (skipped) during the concatenation.
```
[the remark this answers is not in the document]
Comment (sperlingxx) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Fixed

**Comment by sperlingxx** [comment]
Hi @jlowe, thanks for reviewing. I've addressed the problems you pointed out. And I made similar modifications on `ColumnVector.stringConcatenate`.

**Comment by sperlingxx** [comment]
I think we should NOT merge this PR until the native [fix](https://github.com/rapidsai/cudf/pull/8180) getting merged.

**Review by jlowe — APPROVED** [review]

**Pull request merged**